src/main/java/org/javarosa/core/model/TriggerableDag.java

```diff
@@ -59,32 +59,12 @@
 
     protected final EventNotifierAccessor accessor;
 
-    /**
-     * NOT VALID UNTIL finalizeTriggerables() is called!!
-     * <p>
-     * Topologically ordered list, meaning that for any tA and tB in
```

Comment:
Although I do like the new naming and find it helpful, I still find these docs (or a more concise version) helpful.

Reply:
This is related to the issue that the lifecycle of the DAG is still not explicit on its API. 

We currently have to call certain methods in a certain sequence in order to have a functional DAG object. That's why this member is not `final` and we needed the comment in the first place.

I'd rather have a builder object of sorts to manage the DAG's creation, implement its lifecycle in an explicit way and have a completely immutable DAG object in return.

For now, I thought that the use of `DAG` in the member's name could tell everything that has to be known about the contents and ordering of the list, and the member being not `final` and assigning an `emptySet()` could tell about it not actually being usable before something else happens.

Reply:
> I'd rather have a builder object of sorts to manage the DAG's creation, implement its lifecycle in an explicit way and have a completely immutable DAG object in return.

This would also help us separate the contexts where a `Triggerable` should be used as opposed to a `QuickTriggerable`. The former should be used during DAG building, the latter only when the DAG is stable.

Reply:
All that sounds good but I wasn't so much talking about the lifecycle warning as the brief descriptions. The naming helps but I like to read some prose about what fields represent to get a feel for what a class does. That certainly is a personal preference.

Reply:
Understood. I'll add some comments then :)

Reply:
Thanks, I find the comments very helpful. 😊